### PR TITLE
Fix Ganesha's handling of UDP messages < 16 bytes long

### PR DESCRIPTION
**Pull request by chalcogen**
When Ganesha receives a message less than 16 bytes in length, it causes Ganesha to stop rearming the UDP socket afterwards and thus stops processing message on that socket. This change merely moves around code so that the rearm happens before the callback exits.

### CODE REVIEW

File: src/svc_dg.c [hunk 1]
```diff
@@ -279,6 +270,22 @@ svc_dg_rendezvous(SVCXPRT *xprt)
 		return (XPRT_DIED);
 	}
 
+        if (sp->sa_family == (sa_family_t) 0xffff) {
+                __warnx(TIRPC_DEBUG_FLAG_ERROR,
+                        "%s: Bad message sa_family is 0xffff",
+                        __func__);
+                svc_dg_xprt_free(su);
```
Comment (madhuthorat) [inline src/svc_dg.c]: This line should be removed, right ? We don't want to free UDP xprt when the issue happens, as we want UDP xprt to be available after issue happens. @dang what do you think ?

File: src/svc_dg.c [hunk 2]
```diff
@@ -279,6 +270,22 @@ svc_dg_rendezvous(SVCXPRT *xprt)
...
+                return (XPRT_DIED);
+        }
+
+        if (rlen == -1 || (rlen < (ssize_t) (4 * sizeof(u_int32_t)))) {
+                __warnx(TIRPC_DEBUG_FLAG_ERROR,
+                        "%s: Bad message rlen: %d",
+                        __func__, rlen);
+                svc_dg_xprt_free(su);
```
Comment (madhuthorat) [inline src/svc_dg.c]: Same comment as above.

File: src/svc_dg.c (hunk 1 above)
Comment (dang) [reply src/svc_dg.c]: I agree.  The point is to keep the xprt available for use after the bad message, so we should not be freeing the xprt.  This means XPRT_DIED is probably not the correct return, either (although the caller ignores the return value...)
This should probably return SVC_STAT(xprt);

**Review by dang — CHANGES_REQUESTED** [review]

File: src/svc_dg.c (hunk 1 above)
Comment (chalcogen) [reply src/svc_dg.c]: Thanks @madhuthorat @dang updated as requested.

**Comment by dang** [comment]
Can you squash these commits together, rather than having 2 commits?

**Comment by chalcogen** [comment]
Done squashing. Thanks.

**Review by dang — APPROVED** [review]

**Comment by dang** [comment]
The commit needs to be signed.  You can sign with:
git commit -s --amend

(And in general, commit with "git commit -s")

**Comment by chalcogen** [comment]
@dang Done. Thank you for pointing that out.

**Pull request merged**